### PR TITLE
Add multiple choice with completion for shell init files.

### PR DESCRIPTION
**Pull request by jiegec**
Note: using cl-lib functions.

### CODE REVIEW

File: crux.el [hunk 1]
```diff
@@ -352,13 +352,13 @@ Doesn't mess with special buffers."
...
+                           ((string= "bash" shell) '("$BASH_ENV" "$HOME/.bashrc" "$HOME/.bash_profile" "$HOME/.bash_login" "$HOME/.profile" "$HOME/.bash_logout" "/etc/bashrc" "/etc/bash_profile" "/etc/bash_login" "/etc/profile" "/etc/bash_logout"))
+                           ((string= "tcsh" shell) '("$HOME/.login" "$HOME/.cshrc" "$HOME/.tcshrc" "$HOME/.logout" "/etc/csh.cshrc" "/etc/csh.login" "/etc/csh.logout"))
+                           ((string= "fish" shell) '("$HOME/.config/fish/config.fish" "$XDG_CONFIG_HOME/fish/config.fish"))
+                           ((string-prefix-p "ksh" shell) '("$HOME/.profile" "$ENV" "/etc/profile"))
                            (t (error "Unknown shell")))))
-    (find-file-other-window (expand-file-name shell-init-file (getenv "HOME")))))
+    (find-file-other-window (completing-read "Choose shell init file:" (remove-if-not 'file-exists-p (mapcar 'substitute-in-file-name shell-init-file))))))
```
Comment (bbatsov) [inline crux.el]: This should be `cl-remove-if-not`.

File: crux.el [hunk 2]
```diff
@@ -352,13 +352,13 @@ Doesn't mess with special buffers."
   (interactive)
   (let* ((shell (car (reverse (split-string (getenv "SHELL") "/" t))))
          (shell-init-file (cond
-                           ((string= "zsh" shell) ".zshrc")
-                           ((string= "bash" shell) ".bashrc")
-                           ((string= "tcsh" shell) ".login")
-                           ((string= "fish" shell) ".config/fish/config.fish")
-                           ((string-prefix-p "ksh" shell) ".profile")
+                           ((string= "zsh" shell) '("$HOME/.zshrc" "$HOME/.zlogin" "$HOME/.zprofile" "$HOME/.zshenv" "$HOME/.zlogout" "/etc/zshenv" "/etc/zprofile" "/etc/zshrc" "/etc/zlogin" "/etc/zlogout" "$ZDOTDIR/.zshrc" "$ZDOTDIR/.zlogin" "$ZDOTDIR/.zprofile" "$ZDOTIR/.zshenv" "$ZDOTDIR/.zlogout"))
```
Comment (bbatsov) [inline crux.el]: Maybe those long lists should be extracted into `defvar`s?

File: crux.el (hunk 1 above)
Comment (bbatsov) [inline crux.el]: I'd add a check here to do the completing read only if we have at least 2 candidates (which will probably be the case pretty often, but still).

File: crux.el (hunk 2 above)
Comment (bbatsov) [inline crux.el]: Or even better - defcustoms.

**Comment by jiegec** [comment]
Working on it.

**Comment by jiegec** [comment]
Done.

**Comment by bbatsov** [comment]
Looks good. Just squash the two commits together and we're good to go.

**Comment by jiegec** [comment]
Btw, preceding-sexp in an obsolete function as of 25.1, and it suggests using elisp--preceding-sexp instead.

**Comment by bbatsov** [comment]
> Btw, preceding-sexp in an obsolete function as of 25.1, and it suggests using elisp--preceding-sexp instead.

We should check when `elisp--preceding-sepx` was introduced before making any changes.

**Pull request merged**